### ISSUE TITLE
Use a minimal base image Docker images

### ISSUE THREAD

**Issue by wallrj** [sky-uk/cassandra-operator#145]
In https://github.com/sky-uk/cassandra-operator/pull/98/files#r289082025 @alangibson01 wrote:
> Could we look at going with a base image a bit lighter for this sidecar (maybe Alpine, which is ~5MB rather than ~183MB), and a simple init system like https://github.com/Yelp/dumb-init?

@wallrj replied:
> In theory, we could use Scratch as the base image if we enable Shared PID namespaces:
>   https://www.ianlewis.org/en/almighty-pause-container
>   https://medium.com/@chemidy/create-the-smallest-and-secured-golang-docker-image-based-on-scratch-4752223b7324

Let's also consider using https://github.com/GoogleContainerTools/distroless

**Comment by alangibson01** [comment]
Yes, I think distroless and multi-stage builds is probably a good approach to take. For multi-stage builds we'll need to make sure that our Travis build is using a version of Docker that supports them.
